src/features/user/MFV2/ProfileCard/ShareModal/index.tsx

```diff
@@ -0,0 +1,147 @@
+    const subject = t('shareFeature.shareTextTitle');
+    const body = encodeURIComponent(
+      `${t('shareFeature.textToShare', {
+        name: userProfile?.displayName,
+      })}\n\n${linkToShare}`
+    );
+    const mailtoLink = `mailto:?subject=${subject}&body=${body}`;
+
+    window.open(mailtoLink);
+  };
+
+  return (
+    <Modal
+      className={'modalContainer' + ' ' + theme}
+      open={shareModalOpen}
+      onClose={handleShareModalClose}
+      closeAfterTransition
+      aria-labelledby="simple-modal-title"
+      aria-describedby="simple-modal-description"
+      style={{ backdropFilter: 'blur(5px)' }}
+    >
+      <Fade in={shareModalOpen}>
+        <div className={styles.shareModalPopup}>
+          <div className={styles.socialMediaIconContainer}>
+            <h3>{t('shareFeature.shareVia')}</h3>
+            <div>
+              <button
+                onClick={() =>
+                  handleShare(
+                    `https://www.facebook.com/sharer.php?u=${linkToShare}&quote=${textToShareLinkedin}&hashtag=%23StopTalkingStartPlanting`
+                  )
+                }
+                className={styles.socialMediaIcon}
+              >
+                <FacebookCustomIcon />
+              </button>
+              <button
+                onClick={() =>
+                  handleShare(
+                    `https://www.linkedin.com/sharing/share-offsite/?&url=${linkToShare}`
+                  )
+                }
+                className={styles.socialMediaIcon}
+              >
+                <LinkedInCustomIcon />
+              </button>
+
+              <button
+                onClick={() =>
+                  handleShare(`whatsapp://send?text=${linkToShare}`)
+                }
+                className={styles.socialMediaIcon}
+              >
+                <WhatsappCustomIcon />
+              </button>
+
+              <button
+                onClick={() =>
+                  handleShare(
+                    `https://twitter.com/intent/tweet?hashtags=StopTalkingStartPlanting,TrillionTrees&via=trilliontrees&url=${linkToShare}&text=${textToShare}`
+                  )
+                }
+                className={styles.socialMediaIcon}
+              >
+                <XCustomIcon />
+              </button>
+              <button
+                onClick={handleMailShare}
+                className={styles.socialMediaIcon}
+              >
+                <MailCustomIcon />
+              </button>
+            </div>
+          </div>
+          <div className={styles.shareLinkContainer}>
+            <TextField
+              id="share-profile-link"
+              name="share-profile-link"
+              InputProps={{
+                readOnly: true,
+              }}
+              value={linkToShare}
+            />
+            <CopyToClipboard
+              isButton={false}
+              text={linkToShare}
+              customCopyButton={<CopyIcon />}
+            />
```

Comment:
Why do we need a `customCopyButton` here? And why name it `CopyIcon` when it isn't an icon?

Reply:
`customCopyButton` was created so we could pass custom components thus widening the usability of this component. This would also allow us to style the button as per the requirement. Once we move forward with making buttons common, we could discuss about this button as well.